pkg/api/v2/types.go

```diff
@@ -603,17 +605,55 @@ type EKSClusterSpec struct {
 
 	ResourcesVpcConfig VpcConfigRequest `json:"vpcConfigRequest" required:"true"`
 
+	// The Kubernetes network configuration for the cluster.
+	KubernetesNetworkConfig *KubernetesNetworkConfigResponse `json:"kubernetesNetworkConfig,omitempty"`
+
 	// The desired Kubernetes version for your cluster. If you don't specify a value
 	// here, the latest version available in Amazon EKS is used.
-	Version string `json:"version"`
+	Version string `json:"version,omitempty"`
+
+	// The Unix epoch timestamp in seconds for when the cluster was created.
+	CreatedAt *time.Time `json:"createdAt,omitempty"`
+
+	// The metadata that you apply to the cluster to assist with categorization
+	// and organization. Each tag consists of a key and an optional value. You define
+	// both. Cluster tags do not propagate to any other resources associated with
+	// the cluster.
+	Tags map[string]*string `json:"tags,omitempty"`
 
 	// The Amazon Resource Name (ARN) of the IAM role that provides permissions
 	// for the Kubernetes control plane to make calls to AWS API operations on your
 	// behalf. For more information, see Amazon EKS Service IAM Role (https://docs.aws.amazon.com/eks/latest/userguide/service_IAM_role.html)
 	// in the Amazon EKS User Guide .
 	//
 	// RoleArn is a required field
-	RoleArn string `json:"roleArn" required:"true"`
+	RoleArn string `json:"roleArn,omitempty" required:"true"`
```

Comment:
Is it fine to have omitempty and required set to `true` at the same time?

Reply:
yeah, i know its confusing.
the reason is "required" for creation/POST request but `omitempty` for get response